### PR TITLE
[DPG-946] Adding iFIX ES Pipeline service to egovernments ifix repo

### PR DESCRIPTION
**Pull request by shashwat-egov**

### CODE REVIEW

File: domain-services/ifix-es-pipeline/.mvn/wrapper/maven-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.8.4/apache-maven-3.8.4-bin.zip
```
Comment (rushang7-eGov) [inline domain-services/ifix-es-pipeline/.mvn/wrapper/maven-wrapper.properties:1]: Remove the complete `.mvn` directory. It is not required.

File: domain-services/ifix-es-pipeline/LOCALSETUP.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Local Setup
+
+To setup the national-dashboard-kafka-pipeline service in your local system, clone the [Core Service repository](https://github.com/egovernments/core-services).
```
Comment (rushang7-eGov) [inline domain-services/ifix-es-pipeline/LOCALSETUP.md:3]: Name correction on this line and last line

File: domain-services/ifix-es-pipeline/mvnw [hunk 3]
```diff
@@ -0,0 +1,316 @@
+#!/bin/sh
```
Comment (rushang7-eGov) [inline domain-services/ifix-es-pipeline/mvnw:1]: Unnecessary file. Please delete it.

File: domain-services/ifix-es-pipeline/mvnw.cmd [hunk 4]
```diff
@@ -0,0 +1,188 @@
+@REM ----------------------------------------------------------------------------
```
Comment (rushang7-eGov) [inline domain-services/ifix-es-pipeline/mvnw.cmd:1]: Unnecessary file

File: domain-services/ifix-es-pipeline/pom.xml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-beans</artifactId>
+			<version>5.2.20.RELEASE</version>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-web</artifactId>
+		</dependency>
+		<dependency>
```
Comment (rushang7-eGov) [inline domain-services/ifix-es-pipeline/pom.xml:29]: Is this project using DB? If not, then please remove JDBC and Flyway dependencies and corresponding configurations from `application.properties`.

File: domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+     * @param topic
+     */
+    @KafkaListener(topics = { "${fiscal.event.kafka.push.topic}", "${fiscal.event.migration.origin.push.topic}"})
+    public void listen(HashMap<String, Object> record, @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        try {
+            FiscalEventRequest incomingData = objectMapper.convertValue(record, FiscalEventRequest.class);
+
+            // Enrich hierarchy map according to the tenantid encountered by this pipeline to avoid redundant network calls
+            if(!tenantIdVsExpenditureTypeVsUuidsMap.containsKey(incomingData.getFiscalEvent().getTenantId()))
+                tenantIdVsExpenditureTypeVsUuidsMap.put(incomingData.getFiscalEvent().getTenantId(), loadExpenditureTypeVsUuidMap(incomingData.getFiscalEvent().getTenantId()));
```
Comment (rushang7-eGov) [inline domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java:75]: You can write `loadExpenditureTypeVsUuidMap` method as `@PostConstruct`. Then you wouldn't need to write this logic in the listen method.

File: domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+            else
+                expenditureTypeVsUuidsMap.get("Others").add(chartOfAccount.getId());
+
+        });
+        return expenditureTypeVsUuidsMap;
+    }
+
+    private StringBuilder getIfixMasterDataUri() {
+        StringBuilder uri = new StringBuilder(ifixMasterDataServiceHost);
+        uri.append(ifixMasterDataServiceSearchEndpoint);
```
Comment (rushang7-eGov) [inline domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java:113]: This isn't a common master data search endpoint. 
This is a COA search endpoint, so it is better to name the method and variables that include `coa`.

File: domain-services/ifix-es-pipeline/.mvn/wrapper/maven-wrapper.properties (hunk 1 above)
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/.mvn/wrapper/maven-wrapper.properties:1]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/LOCALSETUP.md (hunk 2 above)
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/LOCALSETUP.md:3]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/mvnw (hunk 3 above)
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/mvnw:1]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/mvnw.cmd (hunk 4 above)
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/mvnw.cmd:1]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/pom.xml (hunk 5 above)
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/pom.xml:29]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/config/IESPConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+import javax.annotation.PostConstruct;
+import java.util.TimeZone;
+
+@Component
+@Data
+@Import({TracerConfiguration.class})
+@NoArgsConstructor
+@AllArgsConstructor
+public class IESPConfiguration {
```
[the remark this answers is not in the document]
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/config/IESPConfiguration.java:24]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private ServiceRequestRepository serviceRequestRepository;
+
+    @Autowired
+    private Producer producer;
+
+    @Value("${fiscal.event.es.push.topic}")
```
[the remark this answers is not in the document]
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java:39]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/models/ProducerPOJO.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.util.List;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@ToString
+public class ProducerPOJO {
```
[the remark this answers is not in the document]
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/models/ProducerPOJO.java:16]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/models/ProducerPOJO.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+    @JsonProperty("RequestInfo")
+    private RequestInfo requestInfo;
+
+    @JsonProperty("Records")
+    private List<FiscalEvent> records;
```
[the remark this answers is not in the document]
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/models/ProducerPOJO.java:22]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/service/FiscalDataEnrichmentService.java [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+@Service
+public class FiscalDataEnrichmentService {
+
+    @Autowired
+    private ServiceRequestRepository serviceRequestRepository;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Value("${ifix.department.entity.service.host}")
```
[the remark this answers is not in the document]
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/service/FiscalDataEnrichmentService.java:27]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/src/main/resources/application.properties [hunk 13]
```diff
@@ -0,0 +1,60 @@
+server.contextPath=/ifix
+server.servlet.context-path=/ifix
+server.port=8283
+app.timezone=UTC
+
+# DB CONNECTION CONFIGURATIONS
```
[the remark this answers is not in the document]
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/src/main/resources/application.properties:6]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/src/test/java/org/egov/nationaldashboardkafkapipeline/NationalDashboardKafkaPipelineApplicationTests.java [hunk 14]
```diff
@@ -0,0 +1,13 @@
+package org.egov.nationaldashboardkafkapipeline;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class NationalDashboardKafkaPipelineApplicationTests {
```
[the remark this answers is not in the document]
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/src/test/java/org/egov/nationaldashboardkafkapipeline/NationalDashboardKafkaPipelineApplicationTests.java:7]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java (hunk 6 above)
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java:75]: we don't have the list of tenantIds at the time of loading, that is why we cannot add it to @PostConstruct method.

File: domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java (hunk 7 above)
Comment (shashwat-egov) [reply domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java:113]: minor change, to be taken up as part of code cleanup.

File: domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (rushang7-eGov) [reply domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java]: Pending

File: domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/service/FiscalDataEnrichmentService.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (rushang7-eGov) [reply domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/service/FiscalDataEnrichmentService.java:27]: Pending

**Review by rushang7-eGov — APPROVED** [review]

**Pull request merged**